src/commands/actor/charge.ts

```diff
@@ -0,0 +1,80 @@
+import { APIFY_ENV_VARS } from '@apify/consts';
+import { Args, Flags } from '@oclif/core';
+
+import { getApifyClient } from '../../lib/actor.js';
+import { ApifyCommand } from '../../lib/apify_command.js';
+
+/**
+ * This command can be used to charge for a specific event in the pay-per-event Actor run.
+ * - If run locally or with the --test-pay-per-event flag, it will only log the charge request, without actually charging.
+ * - If run in the Actor run on Apify platform without the testing flag, it will charge the specified amount of events.
+ *
+ * Future TODOs:
+ * - Add logic to work with the max charge USD to prevent exceeding the charging limit.
+ * - Add logic to store events in the log dataset for later inspection to aid local development.
+ */
+export class ChargeCommand extends ApifyCommand<typeof ChargeCommand> {
+	static override description = 'Charge for a specific event in the pay-per-event Actor run.\n';
+
+	static override args = {
+		eventName: Args.string({
+			description: 'Name of the event to charge for',
+			required: true,
+		}),
+	};
+
+	static override flags = {
+		'count': Flags.integer({
+			description: 'Number of events to charge',
+			required: false,
+			default: 1,
+		}),
+		'idempotency-key': Flags.string({
+			description: 'Idempotency key for the charge request',
+			required: false,
+		}),
+		'test-pay-per-event': Flags.boolean({
+			description: 'Test pay-per-event charging without actually charging',
+			required: false,
+			default: false,
+		}),
+	};
+
+	async run() {
+		const { args, flags } = await this.parse(ChargeCommand);
+
+		const { eventName } = args;
+		const { count } = flags;
+		const testPayPerEvent = flags['test-pay-per-event'];
+		const idempotencyKey = flags['idempotency-key'];
```

Comment:
This can be replaced by extracting the args from `this.args`, and flags from `this.flags`